### PR TITLE
coqPackages.http: init at 0.2.1

### PR DESCRIPTION
**Pull request by proux01**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by proux01** [comment]
@vbgl successfully tested on https://github.com/coq-community/coq-nix-toolbox/pull/250

File: pkgs/development/ocaml-modules/menhir/lib.nix [hunk 1]
```diff
@@ -1,4 +1,6 @@
-{ lib, buildDunePackage, coqPackages, version ? "20231231" }:
+{ lib, buildDunePackage, coqPackages, version ? "20240715" }:
+# for some packages such as coqPackages.json, the default version
+# above must be kept in sync with the default version in coqPackages.MenhirLib
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/menhir/lib.nix:3]: This comment requires too much (and unnecessary) coupling between ocamlPackages and coqPackages. I’d recommend to put this constraint where it belongs, i.e., according to this comment in the `coqPackages.json` derivation.

File: pkgs/development/ocaml-modules/menhir/lib.nix [hunk 2]
```diff
@@ -1,4 +1,4 @@
-{ lib, buildDunePackage, coqPackages, version ? "20231231" }:
+{ lib, buildDunePackage, coqPackages, version ? "20240715" }:
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/menhir/lib.nix:2]: This update is wrong.

File: pkgs/development/ocaml-modules/menhir/lib.nix (hunk 1 above)
Comment (proux01) [reply pkgs/development/ocaml-modules/menhir/lib.nix]: Makes sense. In fact, it's a nightmare to get the versions to align whereas it doesn't seem to be fundamentally required. Let's just copy what CompCert is doing.
If https://github.com/coq-community/coq-nix-toolbox/pull/250 confirm this works, we should be good.

File: pkgs/development/ocaml-modules/menhir/lib.nix (hunk 2 above)
Comment (proux01) [reply pkgs/development/ocaml-modules/menhir/lib.nix]: Fixed (I mean, removed the edit)

**Comment by proux01** [comment]
Thanks for your patience!

**Pull request merged**